data-processing/tests/common/commands/test_fetch_arxiv_sources.py

```diff
@@ -0,0 +1,65 @@
+from dataclasses import dataclass
+from typing import List, Optional
+from unittest.mock import call, patch
+
+import pytest
+
+from common.commands.fetch_arxiv_sources import (
+    BACKOFF_FETCH_DELAY,
+    DEFAULT_FETCH_DELAY,
+    FetchArxivSources,
+    MAX_FETCH_ATTEMPTS
+)
+from common.fetch_arxiv import FetchFromArxivException
+from scripts.commands import run_command
+
+
+@dataclass
+class Args:
+    arxiv_ids: List[str]
+    arxiv_ids_file: Optional[str]
+    source: str
+    s3_bucket: Optional[str]
+
+
+def test_makes_up_to_k_attempts_to_fetch_from_arxiv():
+    with patch("common.commands.fetch_arxiv_sources.time.sleep") as mock_sleep:
```

Comment:
When sleep is mocked, does it mean it no longer pauses (i.e., returns instantaneously)?

We have an annotation `@pytest.mark.slow` for slow-running tests that might be appopriate here if there is still some latency associated with the `time.sleep` call.

Reply:
Yeah this overrides the built-in function with a no-op that returns instantly.